src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -399,21 +405,34 @@ findNeighborRanks( std::vector< vtkBoundingBox > boundingBoxes )
  * @param[in] loadedMesh the mesh that was loaded on one or several MPI ranks
  * @param[in] comm the MPI communicator
  * @param[in] partitionRefinement number of graph partitioning refinement cycles
+ * @param[in] useGlobalIds if set to one, use global id arrays from the vtk mesh when available. Else, always recompute them.
  */
 vtkSmartPointer< vtkDataSet >
 redistributeMesh( vtkDataSet & loadedMesh,
                   MPI_Comm const comm,
                   VTKMeshGenerator::PartitionMethod const method,
-                  int const partitionRefinement )
+                  int const partitionRefinement,
+                  int const useGlobalIds )
 {
   GEOSX_MARK_FUNCTION;
 
-  // Generate global IDs for vertices and cells
-  vtkNew< vtkGenerateGlobalIds > generator;
-  generator->SetInputDataObject( &loadedMesh );
-  generator->Update();
-  vtkSmartPointer< vtkDataSet > mesh =
-    vtkDataSet::SafeDownCast( generator->GetOutputDataObject( 0 ) );
+  // Generate global IDs for vertices and cells, if needed
+  vtkSmartPointer< vtkDataSet > mesh;
+  if( useGlobalIds != 1
```

Comment:
Please be more C++ friendly :)
A value of 2 is supposed to be `true` and this is not how you're dealing with it.
Maybe change the signature to `bool const useGlobalIds` instead of `int const useGlobalIds` and properly deal with conversion (or let C++ do it) at call?

Reply:
With latest comit, we have 3 possibilities (<0, 0, >0) so I have kept the flag as an `int`, but it is now handled in a more C++ friendly way :-)